lib/view_component/compiler.rb

```diff
@@ -16,6 +16,7 @@ class Compiler
     def initialize(component_class)
       @component_class = component_class
       @redefinition_lock = Mutex.new
+      @variants_rendering_templates = []
```

Comment:
```suggestion
      @variants_rendering_templates = Set.new()
```

Thoughts on using a Set for faster lookups (although I imagine it's minor improvements but may add up?)?

Reply:
Sure, sounds good to me!